### PR TITLE
add a token container with serialization as CARv1 file

### PR DESCRIPTION
**Pull request by MichaelMure**
Rendered explanations: https://github.com/ucan-wg/go-ucan/blob/40639b67152cc1c6c8b564b700838e2af78b6f57/pkg/container/Readme.md

### CODE REVIEW

File: pkg/container/car.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+// It can be used to have at least one root in a CARv1 file (making it legal), yet
+// denote that it can be ignored.
+var EmptyCid = cid.MustParse([]byte{01, 55, 00, 00})
+
+type carBlock struct {
+	c    cid.Cid
+	data []byte
+}
+
+// writeCar writes a CARv1 file with no roots, containing the blocks from the iterator.
```
Comment (fabiobozzo) [inline pkg/container/car.go:32]:
```suggestion
// writeCar writes a CARv1 file with optional roots, containing the blocks from the iterator.
```
?

File: pkg/container/car.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+func writeCar(w io.Writer, roots []cid.Cid, blocks iter.Seq[carBlock]) error {
+	if len(roots) == 0 {
+		roots = []cid.Cid{EmptyCid}
+	}
+	h := carHeader{
+		Roots:   roots,
+		Version: 1,
+	}
+	hb, err := cbor.DumpObject(h)
```
Comment (fabiobozzo) [inline pkg/container/car.go:41]: is the codec always cbor?

Comment (MichaelMure) [reply pkg/container/car.go:41]: yes, it's part of the CARv1 spec

**Comment by MichaelMure** [comment]
> why CAR v1 instead of v2?

CARv2 is a CARv1 wrapped with an index for fast lookup. We really don't need it. We just need a barebone container from which we'll read every values.

**Pull request merged**